packages/starlight/utils/navigation.ts

```diff
@@ -368,6 +363,37 @@ export function getSidebarFromConfig(
 	}
 }
 
+/** Transform an intermediate sidebar into a sidebar for the current page. */
+function getSidebarFromIntermediateSidebar(
+	intermediateSidebar: SidebarEntry[],
+	pathname: string
+): SidebarEntry[] {
+	const sidebar = structuredClone(intermediateSidebar);
+	setIntermediateSidebarCurrentEntry(sidebar, pathname);
+	return sidebar;
+}
+
+/** Marks the current page as such in an intermediate sidebar. */
+function setIntermediateSidebarCurrentEntry(
+	intermediateSidebar: SidebarEntry[],
+	pathname: string
+): boolean {
+	let found = false;
+	for (const entry of intermediateSidebar) {
+		if (entry.type === 'link') {
+			found = pathsMatch(encodeURI(entry.href), pathname);
+			if (found) {
+				entry.isCurrent = true;
+				break;
+			}
+		} else if (entry.type === 'group') {
+			found = setIntermediateSidebarCurrentEntry(entry.entries, pathname);
+			if (found) break;
+		}
+	}
+	return found;
+}
```

Comment:
Since we are returning a boolean we can use it inside an if statement, this should clean up the code a bit.

I ran the tests locally and seems to be working, feel free to disregard.
```suggestion
function setIntermediateSidebarCurrentEntry(
	intermediateSidebar: SidebarEntry[],
	pathname: string
): boolean {
	for (const entry of intermediateSidebar) {
		if (entry.type === 'link' && pathsMatch(encodeURI(entry.href), pathname)) {
			entry.isCurrent = true;
			return true;
		}

		if (entry.type === 'group' && setIntermediateSidebarCurrentEntry(entry.entries, pathname)) {
			return true;
		}
	}
	return false;
}
```